exchanges/gateio/gateio_ws_futures.go

```diff
@@ -651,19 +657,24 @@ func (g *Gateio) processFuturesOrdersPushData(data []byte, assetType asset.Item)
 	}{}
 	err := json.Unmarshal(data, &resp)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	orderDetails := make([]order.Detail, len(resp.Result))
 	for x := range resp.Result {
-		status, err := order.StringToOrderStatus(func() string {
-			if resp.Result[x].Status == "finished" {
-				return "cancelled"
+		var status order.Status
+		if resp.Result[x].Status == "finished" {
+			if resp.Result[x].FinishAs == "ioc" || resp.Result[x].FinishAs == "reduce_only" {
+				status = order.Cancelled
+			} else {
+				status, err = order.StringToOrderStatus(resp.Result[x].FinishAs)
 			}
-			return resp.Result[x].Status
-		}())
+		} else {
+			status, err = order.StringToOrderStatus(resp.Result[x].Status)
+		}
 		if err != nil {
-			return err
+			return nil, err
```

Comment:
Would appreciate this becoming an order.Classification error instead of a killer. Not high priority or requirement. Just allows for orders to still be known despite any issues on our end